opentracing_instrumentation/client_hooks/__init__.py

```diff
@@ -31,7 +31,7 @@
 import six
 
 
-def install_all_patches():
+def install_all_patches(requests_response_handler_hook=None):
```

Comment:
I am not a fan of this config approach. First, this locks a positional API which might break for people in the future, so I think a kwarg would be better (even though less type safe). Second, the config methods are static anyway, so the configuration can be static too, e.g. `requests.set_response_handler_hook()`. If we don't like static, then the patcher functionality can be refactored into a class, with an optional global instance used by static `install_patches()` function.

Reply:
Also, how does this compare to #36 ?

Reply:
> I am not a fan of this config approach. First, this locks a positional API which might break for people in the future, so I think a kwarg would be better (even though less type safe).

Sorry, I don't understand your concern here since `requests_response_handler_hook` is already a named argument with `None` as the default value (isn't it a "kwarg"?).
Can you please show me some snippet of what do you mean?

> Second, the config methods are static anyway, so the configuration can be static too, e.g. `requests.set_response_handler_hook()`. If we don't like static, then the patcher functionality can be refactored into a class, with an optional global instance used by static `install_patches()` function.

I'll try to refactor it a bit.

> Also, how does this compare to #36 ?

Unfortunately, current implementation of interceptors looks useless in case when we have to deal with responses.
**Do you want me to add some `after_http_request` method just like [before_http_request](https://github.com/uber-common/opentracing-python-instrumentation/blob/576d3eee7fcf99ccdfc76264c91c9c3ac38c0482/opentracing_instrumentation/http_client.py#L35)?**

Reply:
Hello @yurishkuro,
Could you please take a look at the current version?

I can refactor other hooks as well and entirely get rid of current implementation of singleton if you consider it's worth to do so.

Reply:
The kwargs syntax is different from optional arguments. With optional arguments one can still call the function via positional syntax, e.g. `my_func(arg_value)`. Cf. https://www.geeksforgeeks.org/args-kwargs-python/.

The downside of kwargs is that they are completely dynamic and their names cannot be checked by the compiler. Therefore, per my second point, I would rather have a setter somewhere, similar to `install_client_interceptors`

Reply:
Ah, I suppose that I probably got the point.
But I don't really like to do something like
```python
def install_all_patches(*args, **kwargs):
    ...
    if 'requests_response_handler_hook' in kwargs:
        requests_patcher.set_response_handler_hook(
            kwargs['requests_response_handler_hook']
        )
```
I prefer to use `*` or `*args` before `requests_response_handler_hook`, but it's only possible with Python 3.
```python
def install_all_patches(*, requests_response_handler_hook=None):
```
Meanwile, I just removed `requests_response_handler_hook` from `install_all_patches` for now, so I believe we could resolve this discussion.